### PR TITLE
Feature: Add product deletion functionality

### LINKED ISSUE #79: < Nav Menu > Refactor and enhance Navigation Menu
Remove display: fixed from the navigation menu for better layout consistency.
Use prepend to ensure the navigation menu is rendered above other pages.
Delete the line this.container.classList.add("navigation-menu"); from NavigationMenu.js to avoid redundant class additions.
Comment out the line bg.classList.add("background"); in Marketplace.js as it is extraneous.

### LINKED ISSUE #90: < Product Page > Create API Endpoint to Fetch Product by ID
Develop a backend API endpoint that retrieves the details of a single product based on its unique identifier (ID). This endpoint will support the dynamic rendering of the Product Page on the frontend.

### LINKED ISSUE #96: < Product Page > Implement and Test Product Page Backend Functionality
Develop the backend for the product page, including database initialization, Sequelize model definitions, mock data insertion, and API endpoint implementation. Ensure the backend is fully functional and can retrieve mock data for the frontend integration.

### LINKED ISSUE #97: Fix Merge Conflicts Between main and bensonMain
While merging main with bensonMain, there were conflicts that need to be resolved to prioritize changes from bensonMain. The goal is to ensure that all necessary changes from bensonMain are retained while maintaining the integrity of the main branch.

### LINKED ISSUE #99: < Product Page > Render product page from mock data using Sequelize
Updated the product page to dynamically render its elements using mock data stored in the SQLite database via Sequelize.

### LINKED ISSUE #103: < Product Page > Display Product Page and Make Functions Async
We need to implement a feature that displays the product details on the product page. The functionality should fetch product data from the backend (such as product name, description, images, and reviews) and render it on the page. To improve performance and handle asynchronous operations smoothly, we also need to ensure the relevant functions are asynchronous.

Tasks:

Fetch Product Data:
Create an asynchronous function to fetch the product data using the product ID.
The data should include:
  Product name
  Category
  Description
  Price
  Images
  Reviews
  Seller information
  Ensure the data is fetched correctly from the API or backend service.

Product Page Component:
Implement a ProductPage component that will display the fetched product data.

Async Functions:
Refactor any necessary functions (e.g., fetchProductData(), render()) to be async to handle the asynchronous nature of data fetching.
Use await to ensure the data is fully loaded before rendering the page.

### LINKED ISSUE #104: < Product Page > Set up API Endpoint for Getting All Products
We need to implement an API endpoint that will allow users to retrieve all products from the database.

### LINKED ISSUE #105: < Product Page > Set up API Endpoint for Posting a New Product
We need to implement an API endpoint that will allow users to add a new product to the database.

### LINKED ISSUE #106: < Product Page > Set up API Endpoint for Updating a Product
We need to implement an API endpoint that will allow users to update the details of an existing product.

### LINKED ISSUE #107: < Product Page > Set up API Endpoint for Deleting a Product
We need to implement an API endpoint that will allow users to delete a product from the database.

### PR DESCRIPTION
**Pull request by batterydied**
This PR adds a DELETE /product/:id endpoint to the API, allowing products to be deleted from the database. This functionality improves product management by enabling the removal of products as needed.

### CODE REVIEW

File: frontend/main/app/AppController.js [hunk 1]
```diff
@@ -11,64 +7,76 @@ import { NavigationMenu } from '../components/NavigationMenu/NavigationMenu.js';
...
+     // Set the default page as the marketplace
+     this.#currentView = this.#views.marketplace;
    }
 
-  render() {
-    if (!this.#container) {
-      this.#container = document.createElement('div');
-      this.#container.classList.add('app-controller');
-    }
+   async render(id=null) {
```
Comment (wnauda) [inline frontend/main/app/AppController.js:33]: Why is render made asynchronous?

File: frontend/main/components/MarketplacePage/Products.js [hunk 2]
```diff
@@ -2,281 +2,312 @@
...
-      price: 11.99
-    }
+  {
+   // Clownfish
+    prodid: "1a2b3c4d5e",
+    sellername: "Ocean Wonders",
+    sellerid: "1234-5678",
+    imgurl: "https://miro.medium.com/v2/resize:fit:940/1*tqkPCH9gc8nv000FOpRoOA.png",
+    name: "Clownfish",
+    category: Category.SaltwaterFish,
```
Comment (wnauda) [inline frontend/main/components/MarketplacePage/Products.js:293]: To make sorting by categories easier, there are only 5 categories: Fish, Rocks, Plants, Tanks, and Misc, so we will need to update our test data to use those categories.

**Review by wnauda — APPROVED** [review]
Changes look good and work as intended per my testing. I think we can merge this.

File: frontend/main/app/AppController.js (hunk 1 above)
Comment (batterydied) [reply frontend/main/app/AppController.js:33]: Fetching data from DB is an async operation

**Pull request merged**